wgsl/index.bs

```diff
@@ -11874,19 +11881,51 @@ See [[#builtin-inputs-outputs]] for how to specify that a pipeline input or outp
       <td>vertex
       <td>output
       <td>vec4&lt;f32&gt;
-      <td style="width:50%">Output position of the current vertex, using homogeneous coordinates.
-      After homogeneous normalization (where each of the *x*, *y*, and *z* components
-      are divided by the *w* component), the position is in the WebGPU normalized device
-      coordinate space.
-      See [[WebGPU#coordinate-systems]].
+      <td style="width:50%">Output position of the current vertex.
+
+      An output value (*x*,*y*,*z*,*w*)
+      [=behavioral requirement|will=] map to [=clip position=] (*x*/*w*, *y*/*w*, *z*/*w*) in
+      WebGPU [=normalized device coordinates=].
+
+      See [[WebGPU#coordinate-systems]] and [[WebGPU#primitive-clipping]].
 
   <tr>
       <td>fragment
       <td>input
       <td>vec4&lt;f32&gt;
-      <td style="width:50%">Framebuffer position of the current fragment in [=framebuffer=] space.
-      (The *x*, *y*, and *z* components have already been scaled such that *w* is now 1.)
-      See [[WebGPU#coordinate-systems]].
+      <td style="width:50%">
+      <div algorithm="fragment position calculation">
+      Input position of the current fragment.
+
+      Let |fp| be the input position of the fragment.
+      Let |rp| be the [=RasterizationPoint=] for the fragment.
+      Then schematically:
+      <blockquote>
+      |fp|.xy = |rp|.[=rasterizationpoint-destination|destination=].[=fragmentdestination-position|position=]<br>
+      |fp|.z = |rp|.[=rasterizationpoint-depth|depth=]<br>
+      |fp|.w = |rp|.[=rasterizationpoint-perspectivedivisor|perspectiveDivisor=]
+      </blockquote>
+
+      In more detail:
+      *  |fp|.x and |fp|.y are the interpolated x and y coordinates of the
+          position the current fragment in
+          the [=framebuffer=].
+
+          The framebuffer is a two-dimensional grid of pixels with the top-left at (0.0,0.0)
+          and the bottom right at (*vp.width*, *vp.height*).
+          Each pixel has an extent of 1.0 unit in each of the x and y dimensions,
+          and pixel centers are at (0.5,0.5) offset from integer coordinates.
+
+      * |fp|.z is the interpolated depth of the current fragment,
+          in [=normalized device coordinates=], and hence in the range [0.0, 1.0].
```

Comment:
It can be outside that range if unclippedDepth is used. I think this clause can just be removed.

That said I am pretty sure fp.z and rp.depth are not in the same range. IIRC fp.z should be 0-1 and rp.z should be vpmin-vpmax.
https://github.com/gpuweb/cts/blob/c7d833badcd37dc46a999ebeebbbde1368ff15b5/src/webgpu/api/operation/rendering/depth_clip_clamp.spec.ts#L76-L87

Reply:
Got it. (again? Finally? gosh!)   

I think I confirmed with the D3D11 functional spec.  15.6 Viewport, see computation of Z<sub>rt</sub>.  But it doesn't *quite* connect the dots to that also being the z value in the pixel shader.

Vulkan is clearer. 
*  [27.7 Coordinate transformations](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#vertexpostproc-coord-transform) spells out clip coordinates --> NDC --> framebuffer coordinates, where the latter clearly encompasses z.   
      *  Where 'd' is for 'device' as in NDC and 'c' is for 'clip', z<sub>d</sub> = z<sub>c</sub> / w<sub>c</sub>.
* Then [27.9 Controlling the viewport](3-extensions/html/vkspec.html#vertexpostproc-viewport) has NDC --> framebuffer as:
      *  z<sub>f</sub> = p<sub>z</sub> + z<sub>d</sub> + o<sub>z</sub>.  Where in the ordinary case pz - (viewport.maxDepth - viewport.minDepth), and oz = viewport.minDepth.


I think the API spec needs fixing too then.
For rp.depth it says:

> refers to the depth in [NDC](https://gpuweb.github.io/gpuweb/#ndc)

But it should be in that notion of framebuffer coordinates for z.